### DISCUSSION TITLE
get_order_book

### DISCUSSION THREAD

**Discussion by qaplok** [qwertyo1/tinkoff-trading-bot#8] [Q&A] [answered]
Hello. _I try to add function 'get_order_book' to the client, but get error "TypeError: bad argument type for built-in operation".
That is my effort:_ 
async def get_order_book(self, **kwargs) -> GetOrderBookResponse:
        return await self.client.market_data.get_order_book(**kwargs)

_and i call it:_ order_book = await client.get_order_book(figi=[self.figi], depth=50)
_I don't understand what I'm doing wrong, please help.

Regards,
Qaplok_

**Comment by qwertyo1** [comment, marked as answer]
Hello sir!
I am unable to check it right now. I will try it tomorrow.
But as a quick response I can send this screenshot of the documentation. It does not expect figi as an array. Also, figi parameter is deprecated and you should use instrument_id.
![Screenshot_20230814-212636_Chrome](https://github.com/qwertyo1/tinkoff-trading-bot/assets/20947137/fc3db8ca-25f7-4cc8-a2ae-5f15df0f07c5)

Best regards,
Vladimir

**Reply by qaplok** [reply]
Vladimir, I apologize for my inattention. Thank you very much. This seems to be the only thing I missed.

**Reply by qwertyo1** [reply]
No worries! I am glad to help 😊
